tests/Elasticsearch/Tests/YamlRunnerTest.php

```diff
@@ -77,727 +67,621 @@ public static function setUpBeforeClass()
-            $passed = true;
-        } elseif ($caught === 'RequestTimeout408Exception' && $expectedError === 'request_timeout') {
-            $passed = true;
-        } elseif ($caught === 'BadRequest400Exception' && $expectedError === 'request') {
-            $passed = true;
-        } elseif ($caught === 'ServerErrorResponseException' && $expectedError === 'request') {
-            $passed = true;
-        } elseif ($caught === 'RuntimeException' && $expectedError === 'param') {
-            $passed = true;
-        } elseif ($caught === 'Missing404Exception' && $expectedError === 'missing') {
-            $passed = true;
-        }
-
-        if ($passed === false) {
-            if (YamlRunnerTest::checkExceptionRegex($expectedError, $exception)) {
-                $passed = true;
-            } elseif ($exception->getPrevious() !== null) { // try second level
-                if (YamlRunnerTest::checkExceptionRegex($expectedError, $exception->getPrevious())) {
-                    $passed = true;
-                }
-            }
-        }
+        $value = $this->resolveValue($lastOperationResult, key($operation), $context);
+        $expected = current($operation);
 
-        if ($passed === true) {
-            $this->assertTrue(true, $this->log);
-            if ($exception->getPrevious() !== null) {
-                return json_decode($exception->getPrevious()->getMessage(), true);
-            }
-            return json_decode($exception->getMessage(), true);
-        }
+        static::assertCount($expected, $value, 'Failed to gte in test ' . $testName);
 
-        //$this->fail("Tried to match exception, failed.  Exception: ".$exception->getMessage());
-        throw $exception;
+        return $lastOperationResult;
     }
 
-
-    private static function checkExceptionRegex($expectedError, \Exception $exception)
+    /**
+     * Set a variable into context from last operation
+     *
+     * @param $operation
+     * @param $lastOperationResult
+     * @param $context
+     * @param $testName
+     */
+    public function operationSet($operation, $lastOperationResult, &$context, $testName)
     {
-        return isset($expectedError) === true && preg_match("/$expectedError/", $exception->getMessage()) === 1;
+        $key = key($operation);
+        $value = $this->resolveValue($lastOperationResult, $key, $context);
+        $variable = current($operation);
+
+        $context['$' . $variable] = $value;
+
+        return $lastOperationResult;
     }
 
-    private function callMethod($method, $hash, $future)
+    /**
+     * Skip an operation depending on Elasticsearch Version
+     *
+     * @param $operation
+     * @param $lastOperationResult
+     * @param $testName
+     */
+    public function operationSkip($operation, $lastOperationResult, $testName)
     {
-        $ret = array();
+        if (property_exists($operation, 'features')) {
+            // No features supported
```

Comment:
This is likely my fault, as the old method called it a "whitelist", when in reality it was a blacklist.  E.g. a list of features *not* supported.  

We should setup a proper whitelist so that the runner knows what features are supported.  It should support the `stash_in_path` feature to be compatible with the old runner, and soon i'll need to add support for headers and warnings